packages/signatures/src/AbstractSessionSigner.ts

```diff
@@ -103,9 +100,7 @@ export abstract class AbstractSessionSigner<AuthorizationData> implements Sessio
 			const { signer, session } = this.getCachedSession(message.topic, message.payload.address) ?? {}
 			assert(signer !== undefined && session !== undefined)
 
-			// only sign our own current sessions
-			// use a deep comparison
-			assert(deepEquals(message.payload, session))
+			assert(equals(json.encode(message.payload), json.encode(session)))
```

Comment:
There was a reason I switched from comparing `json.encode` to our own `deepEquals` implementation - iirc there was a test that was failing with json.encode comparison; what's the reason for switching back?

Reply:
oh sorry, i misread the git commit log and thought i had added it

the `deepEquals` function fails on Uint8Arrays:

```
> let a = new Uint8Array([0,1,2])
undefined
> let b = new Uint8Array([0,1,2])
undefined
> deepEquals(a,b)
false
```

the reason why the interface tests pass is that we don't have the test case where we have two sessions that have the same data but are different objects in memory

Reply:
I'm happy to just fix the deepEquals function to make this work and add another test case, arguably it should just have its own unit test

Reply:
oh gotcha. let me fix that